modules/calib/test/test_multiview_calib.cpp

```diff
@@ -0,0 +1,186 @@
+                (float)rng.uniform(ty_min, ty_max),
+                (float)rng.uniform(tz_min, tz_max)));
+
+        R.convertTo(R, CV_32F);
+        cv::Mat pattern_new = (R * (pattern - centroid * ones) + centroid * ones  + t * ones).t();
+
+        std::vector<cv::Mat> img_pts_cams(num_cameras);
+        std::vector<bool> visible(num_cameras, false);
+        int num_visible_patterns = 0;
+        for (int c = 0; c < num_cameras; c++) {
+            cv::Mat img_pts;
+            if (is_fisheye[c]) {
+                cv::fisheye::projectPoints(pattern_new, img_pts, Rs_gt[c], Ts_gt[c], Ks_gt[c], distortions_gt[c]);
+            } else {
+                cv::projectPoints(pattern_new, Rs_gt[c], Ts_gt[c], Ks_gt[c], distortions_gt[c], img_pts);
+            }
+
+            // add normal / Gaussian noise to image points
+            cv::Mat noise (img_pts.rows, img_pts.cols, img_pts.type());
+            rng.fill(noise, cv::RNG::NORMAL, 0, noise_std);
+            img_pts += noise;
+
+            bool are_all_pts_in_image = true;
+            const auto * const pts = (float *) img_pts.data;
+            for (int i = 0; i < num_pts; i++) {
+                if (pts[i*2  ] < 0 || pts[i*2  ] > (float)image_sizes[c].width ||
+                    pts[i*2+1] < 0 || pts[i*2+1] > (float)image_sizes[c].height) {
+                    are_all_pts_in_image = false;
+                    break;
+                }
+            }
+            if (are_all_pts_in_image) {
+                visible[c] = true;
+                num_visible_patterns += 1;
+                img_pts.copyTo(img_pts_cams[c]);
+            }
+        }
+
+        if (num_visible_patterns >= 2) {
+            objPoints.emplace_back(board_pattern);
+            visibility.emplace_back(visible);
+            for (int c = 0; c < num_cameras; c++) {
+                image_points_all[c].emplace_back(img_pts_cams[c].clone());
+            }
+            if (objPoints.size() >= MAX_FRAMES)
+                break;
+        }
+    }
+    cv::Mat visibility_mat = cv::Mat_<bool>(num_cameras, (int)objPoints.size());
+    for (int c = 0; c < num_cameras; c++) {
+        for (int f = 0; f < (int)objPoints.size(); f++) {
+            visibility_mat.at<bool>(c, f) = visibility[f][c];
+        }
+    }
+
+    std::vector<cv::Mat> Ks, distortions, Rs, Ts;
+    cv::Mat errors_mat, output_pairs, rvecs0, tvecs0;
+    EXPECT_TRUE(calibrateMultiview (objPoints, image_points_all, image_sizes, visibility_mat,
+       Rs, Ts, Ks, distortions, rvecs0, tvecs0, is_fisheye, errors_mat, output_pairs));
+
+    const double K_err_tol = 1e1, dist_tol = 2e-1, R_tol = 1e-2, T_tol = 1e-2;
+    for (int c = 0; c < num_cameras; c++) {
+        cv::Mat R;
+        cv::Rodrigues(Rs[c], R);
+        EXPECT_MAT_NEAR(Ks_gt[c], Ks[c], K_err_tol);
+        EXPECT_MAT_NEAR(distortions_gt[c], distortions[c], dist_tol);
+        EXPECT_MAT_NEAR(Rs_gt[c], R, R_tol);
+        EXPECT_MAT_NEAR(Ts_gt[c], Ts[c], T_tol);
+    }
+}
```

Comment:
There are stereoCalibrate tests for pinhole and fisheye cameras with real data. I recommend you to re-use their input and ground truth to add tests with real calibration points.
See https://github.com/opencv/opencv/blob/99d216c8d4cc722b8d4387c462c6aec8f61504f4/modules/calib/test/test_fisheye.cpp#L694
and https://github.com/opencv/opencv/blob/99d216c8d4cc722b8d4387c462c6aec8f61504f4/modules/calib/test/test_cameracalibration.cpp#L1242

Reply:
tests are added: https://github.com/opencv/opencv/pull/22363/commits/019915854b897f22478ea76f30a9fe45357adc62